IHP/IDE/FileWatcher.hs

```diff
@@ -13,8 +15,26 @@ withFileWatcher :: (?context :: Context) => IO () -> IO ()
 withFileWatcher inner = withAsync callback \_ -> inner
     where
         callback = FS.withManagerConf fileWatcherConfig \manager -> do
-                FS.watchTree manager "." shouldActOnFileChange handleFileChange
+                watchRootDirectoryFiles manager
+                watchSubDirectories manager
                 forever (threadDelay maxBound) `finally` FS.stopManager manager
+        watchRootDirectoryFiles manager = 
+                FS.watchDir manager "." shouldActOnFileChange handleFileChange 
+        watchSubDirectories manager = do
+                directories <- listWatchableDirectories
+                forM_ directories \directory -> do
+                  FS.watchTree manager directory shouldActOnFileChange handleFileChange
```

Comment:
What happens when a new root directory is added? (e.g. when an new sub application such as `Admin/` is created inside the IHP app)

Reply:
Good point we should also keep track of those and watch them as needed.

Reply:
That's done, anything else you would like to add?